triton-vm/src/table/constraint_circuit.rs

```diff
@@ -49,20 +47,99 @@ impl Display for CircuitId {
     }
 }
 
+pub trait InputIndicator:
+    Debug + Clone + Copy + Hash + PartialEq + Eq + Display + From<usize> + Into<usize>
+{
+}
+
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum SingleRowIndicator<const COLUMN_COUNT: usize> {
+    Row(usize),
+}
+
+impl<const COLUMN_COUNT: usize> Display for SingleRowIndicator<COLUMN_COUNT> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let input_indicator: String = match self {
+            SingleRowIndicator::Row(i) => format!("row[{i}]"),
+        };
+
+        writeln!(f, "{input_indicator}")
+    }
+}
+
+impl<const COLUMN_COUNT: usize> From<usize> for SingleRowIndicator<COLUMN_COUNT> {
+    fn from(val: usize) -> Self {
+        assert!(val < COLUMN_COUNT, "Cannot index out of width of table");
+        SingleRowIndicator::Row(val)
+    }
+}
+
+impl<const COLUMN_COUNT: usize> From<SingleRowIndicator<COLUMN_COUNT>> for usize {
+    fn from(val: SingleRowIndicator<COLUMN_COUNT>) -> usize {
+        match val {
+            SingleRowIndicator::Row(i) => i,
+        }
+    }
+}
+
+impl<const COLUMN_COUNT: usize> InputIndicator for SingleRowIndicator<COLUMN_COUNT> {}
+
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq)]
+pub enum DualRowIndicator<const COLUMN_COUNT: usize> {
+    CurrentRow(usize),
+    NextRow(usize),
+}
+
+impl<const COLUMN_COUNT: usize> Display for DualRowIndicator<COLUMN_COUNT> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let input_indicator: String = match self {
+            DualRowIndicator::CurrentRow(i) => format!("current_row[{i}]"),
+            DualRowIndicator::NextRow(i) => format!("next_row[{i}]"),
+        };
+
+        writeln!(f, "{input_indicator}")
+    }
+}
```

Comment:
This is important for the generated code – add a comment indicating that it is.

Reply:
Replace with custom function.